### DISCUSSION TITLE
Dynamic, updating, distance offset possible?

### DISCUSSION THREAD

**Discussion by chasegiunta** [floating-ui/floating-ui#1444] [General]
Imagine the window below is scrollable. Is it possible to keep the floating element attached to the normal reference point, but outside of a parent container (almost like a dynamic offset), while still possibly utilizing flip when the reference edge is scrolled vertically down outside the container viewport? This isn't my exact use case but similar enough.

![image](https://user-images.githubusercontent.com/1377169/145837629-84096bce-01c7-405c-8575-abf8c9174619.png)

**Comment by atomiks** [comment]
This would be possible with a custom middleware. You'd detect the available space remaining like the `size` middleware does, but use that to offset (in this example) the `y` value by that much so it stays outside of the container. Using `flip` might be odd though as it's outside the boundary, but maybe there are some tricks you could use

https://github.com/floating-ui/floating-ui/blob/5e5764cbebca83f59b3f0e2315f059cefc1f66c6/packages/core/src/middleware/size.ts#L26-L47

**Reply by chasegiunta** [reply]
@atomiks Nice! Thanks for pointing me in the right direction!
